core/src/androidMain/kotlin/BluetoothState.kt

```diff
@@ -1,31 +1,16 @@
 package com.juul.kable
 
 import android.bluetooth.BluetoothAdapter.ACTION_STATE_CHANGED
-import android.bluetooth.BluetoothAdapter.ERROR
-import android.bluetooth.BluetoothAdapter.EXTRA_STATE
-import android.content.BroadcastReceiver
-import android.content.Context
-import android.content.Intent
 import android.content.IntentFilter
+import androidx.core.content.ContextCompat.RECEIVER_NOT_EXPORTED
+import com.juul.tuulbox.coroutines.flow.broadcastReceiverFlow
 import kotlinx.coroutines.DelicateCoroutinesApi
 import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.channels.awaitClose
 import kotlinx.coroutines.flow.SharingStarted.Companion.WhileSubscribed
-import kotlinx.coroutines.flow.callbackFlow
 import kotlinx.coroutines.flow.shareIn
 
 private val intentFilter = IntentFilter(ACTION_STATE_CHANGED)
 
 @OptIn(DelicateCoroutinesApi::class)
-internal val bluetoothState = callbackFlow {
-    val receiver = object : BroadcastReceiver() {
-        override fun onReceive(context: Context, intent: Intent) {
-            val state = intent.getIntExtra(EXTRA_STATE, ERROR)
-            channel.trySend(state)
-        }
-    }
-    applicationContext.registerReceiver(receiver, intentFilter)
-    awaitClose {
-        applicationContext.unregisterReceiver(receiver)
-    }
-}.shareIn(GlobalScope, started = WhileSubscribed(replayExpirationMillis = 0))
+internal val bluetoothState = broadcastReceiverFlow(intentFilter, RECEIVER_NOT_EXPORTED)
```

Comment:
nit: `RECEIVER_NOT_EXPORTED` is default, right?

```suggestion
internal val bluetoothState = broadcastReceiverFlow(intentFilter)
```

Reply:
Tons of conflicting information about this.
`ContextCompat` documentation says system BroadcastReceivers need to be exported:
```
flags – If this receiver is listening for broadcasts sent from the system or from other apps—even other apps that you own—use the RECEIVER_EXPORTED flag. If instead this receiver is listening only for broadcasts sent by your app, use the RECEIVER_NOT_EXPORTED flag.
```
 But Google says that's a security vulnerability and isn't allowed in Android 14 (system receivers should have no flag at all):
https://developer.android.com/about/versions/14/behavior-changes-14#system-broadcasts

Documentation in `Context` seems to be most up to date, so I'm sticking with that:
```
For apps targeting Build.VERSION_CODES.UPSIDE_DOWN_CAKE, either RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED must be specified if the receiver is not being registered for system broadcasts or a SecurityException will be thrown. See registerReceiver(BroadcastReceiver, IntentFilter, int) to register a receiver with flags.
```

Reply:
Maybe you misunderstood my comment? I was saying that we don't really need to specify the parameter because `RECEIVER_NOT_EXPORTED` is the default in Tuulbox: https://github.com/JuulLabs/tuulbox/blob/b592470845918d99e5d6c128243a5038e753fc38/coroutines/src/androidMain/kotlin/flow/BroadcastReceiverFlow.kt#L18

So, I wasn't saying we should not use `RECEIVER_NOT_EXPORTED`, rather: that flag will be used (by Tuulbox) if we omit the parameter?

Reply:
I understood the comment, but it got me looking closer at the documentation since I realized it didn't jive with what Google's Android 14 migration docs said.